crates/orchestrator/src/tests/jobs/da_job/mod.rs

```diff
@@ -1,64 +1,310 @@
-use crate::jobs::Job;
+use crate::tests::common::init_config;
+
+/// Tests the DA Job's handling of a blob length exceeding the supported size.
+/// It mocks the DA client to simulate the environment and expects an error on job processing.
+/// Validates the error message for exceeding blob limits against the expected output.
+/// Asserts correct behavior by comparing the received and expected error messages.
+#[rstest]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/640641.txt",
+    "src/tests/jobs/da_job/test_data/nonces/640641.txt",
+    "640641",
+    110
+)]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/631861.txt",
+    "src/tests/jobs/da_job/test_data/nonces/631861.txt",
+    "631861",
+    110
+)]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/638353.txt",
+    "src/tests/jobs/da_job/test_data/nonces/638353.txt",
+    "63853",
+    110
+)]
+#[tokio::test]
+async fn test_da_job_process_job_failure_on_impossible_blob_length(
+    #[case] state_update_file: String,
+    #[case] noances_file: String,
+    #[case] internal_id: String,
+    #[case] current_blob_length: u64,
+) -> Result<()> {
+    // Mocking DA client calls
+    let mut da_client = MockDaClient::new();
+    // dummy state will have more than 1200 bytes
+    da_client.expect_max_blob_per_txn().with().returning(|| 1);
+    da_client.expect_max_bytes_per_blob().with().returning(|| 1200);
+
+    let server = TestConfigBuilder::new().mock_da_client(Box::new(da_client)).build().await;
+    let config = config().await;
+
+    let state_update = read_state_update_from_file(state_update_file.as_str()).expect("issue while reading");
+
+    let state_update = MaybePendingStateUpdate::Update(state_update);
+    let state_update = serde_json::to_value(&state_update).unwrap();
+    let response = json!({ "id": 640641,"jsonrpc":"2.0","result": state_update });
+
+    get_nonce_attached(&server, noances_file.as_str());
+
+    let state_update_mock = server.mock(|when, then| {
+        when.path("/").body_contains("starknet_getStateUpdate");
+        then.status(200).body(serde_json::to_vec(&response).unwrap());
+    });
+
+    let max_blob_per_txn = config.da_client().max_blob_per_txn().await;
+
+    let response = DaJob
+        .process_job(
+            config.as_ref(),
+            &mut JobItem {
+                id: Uuid::default(),
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String(internal_id.to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
+
+    match response {
+        Ok(_) => {
+            return Err(eyre!("This testcase's process_job was supposed to throw an error, it successed instead."))
+        }
+        Err(e) => {
+            let expected = eyre!(
```

Comment:
To implement `thiserror`  here we would have to make a separate `enum` and with a single `error type`.

```rust 
use thiserror::Error;

#[derive(Error, Debug)]
pub enum CustomError {
    #[error("Exceeded the maximum number of blobs per transaction: allowed {max_allowed}, found {found} for block {block_id} and job id {job_id}")]
    ExceededMaxBlobs {
        max_allowed: usize,
        found: usize,
        block_id: String,
        job_id: Uuid,
    }
}

// Your function that might throw this error
fn test_da_job_process_job_failure_on_small_blob_size() -> Result<()> {
    // ... code logic here ...
    
    Err(CustomError::ExceededMaxBlobs {
        max_allowed: max_blob_per_txn,
        found: current_blob_length,
        block_id: internal_id.to_string(),
        job_id: Uuid::default(),
    })
}
```

1. Is this the correct way to implement the same ?
2. If yes, then doing this (making an enum) for a single error is justified ?

Reply:
1. Pretty much. CustomerError => DaJob Error. If it's a `Job` trait function then we need to create a `JobError` type which will have a `DaError` type. I can show you the details on call/in person 
2. More errors will be added here

Reply:
After in person discussion,
we have moved this update to a separate PR because of large scope of changes.
[PR 66](https://github.com/madara-alliance/madara-orchestrator/pull/66).